src/BigInteger/Adapter/Gmp.php

```diff
@@ -47,8 +47,15 @@ public function init($operand, $base = null)
         set_error_handler(function () {
  /* Do nothing */
         }, \E_WARNING);
-        $res = gmp_init($sign . $operand, $base);
+
+        try {
+            $res = gmp_init($sign . $operand, $base);
+        } catch (\TypeError $e) {
+            $res = false;
+        }
```

Comment:
PHP 8 now throws TypeError instead of issuing a warning. I kept the warning logic for 7.3/7.4.